src/shared/libs/iota/quorum.js

```diff
@@ -0,0 +1,415 @@
+
+                idx += 1;
+            }
+
+            return quorumResult;
+        });
+};
+
+/**
+ *   From a list of synced nodes, compute a quorum result for getBalances iota api
+ *
+ *   @method getQuorumForGetBalances
+ *   @param {array} payload - Addresses
+ *   @param {number} threshold
+ *   @param {array} syncedNodes
+ *
+ *   @returns {Promise}
+ **/
+const getQuorumForGetBalances = (payload, threshold, tips, syncedNodes) => {
+    const requestPayloadSize = size(payload);
+
+    return rejectIfNotEnoughSyncedNodes(syncedNodes)
+        .then(() =>
+            Promise.all(
+                map(
+                    syncedNodes,
+                    (provider) =>
+                        new Promise((resolve) => {
+                            new IOTA({ provider }).api.getBalances(
+                                payload,
+                                threshold,
+                                tips,
+                                (err, states) => (err ? resolve(undefined) : resolve(states)),
+                            );
+                        }),
+                ),
+            ),
+        )
+        .then((results) => {
+            const validResults = filter(results, (result) => !isUndefined(result));
+
+            let idx = 0;
+            const quorumResult = {
+                references: tips,
+                balances: [],
+            };
+
+            while (idx < requestPayloadSize) {
+                /* eslint-disable no-loop-func */
+                quorumResult.balances.push(
+                    determineQuorumResult(map(validResults, ({ balances }) => balances[idx]))(
+                        'getBalances',
+                        QUORUM_THRESHOLD,
+                    ),
+                );
+                /* eslint-enable no-loop-func */
+
+                idx += 1;
+            }
+
+            return quorumResult;
+        });
+};
+
+/**
+ *   For a list of synced nodes, compute a quorum result for latestSolidSubtangleMilestone
+ *
+ *   @method getQuorumForLatestSolidSubtangleMilestone
+ *   @param {array} syncedNodes
+ *
+ *   @returns {Promise}
+ **/
+const getQuorumForLatestSolidSubtangleMilestone = (syncedNodes) => {
+    return rejectIfNotEnoughSyncedNodes(syncedNodes)
+        .then(() =>
+            Promise.all(
+                map(
+                    syncedNodes,
+                    (provider) =>
+                        new Promise((resolve) => {
+                            new IOTA({ provider }).api.getNodeInfo(
+                                (err, info) => (err ? resolve(undefined) : resolve(info)),
+                            );
+                        }),
+                ),
+            ),
+        )
+        .then((results) => {
+            const validResults = filter(results, (result) => !isUndefined(result));
+
+            // Get quorum result for latest solid subtangle milestone
+            const latestSolidSubtangleMilestone = determineQuorumResult(
```

Comment:
nit-pick, but this should follow the structure of the code above, it makes maintenance harder (I understand that there is no payload and the response is a single value)

Reply:
Good suggestion. As mentioned [here](https://github.com/iotaledger/trinity-wallet/pull/631#discussion_r241632229), the pattern is now consistent with these [changes](https://github.com/laumair/trinity-wallet/blob/feat/quorum/src/shared/libs/iota/quorum.js#L180-L284).